src/controllers/emailVault/emailVault.ts

```diff
@@ -441,12 +443,39 @@ export class EmailVaultController extends EventEmitter {
     }
   }
 
-  get banners(): Banner[] {
-    if (!Object.keys(this.emailVaultStates?.email).length) return []
+  get hasKeystoreRecovery() {
+    const keyStoreUid = this.#keyStore.keyStoreUid
+    const EVEmails = Object.keys(this.emailVaultStates.email)
+
+    if (!keyStoreUid || !EVEmails.length) return false
+
+    return !!EVEmails.find((email) => {
+      return (
+        this.emailVaultStates.email[email].availableSecrets[keyStoreUid]?.type ===
+        SecretType.KeyStore
+      )
+    })
+  }
 
+  get banners(): Banner[] {
     const banners: Banner[] = []
 
-    Object.keys(this.emailVaultStates?.email || {}).forEach((email) => {
+    if (!this.hasKeystoreRecovery) {
+      banners.push({
+        id: 'keystore-secret-backup',
+        topic: 'WARNING',
+        title: 'Enable local key store recovery with Email Vault',
+        text: "Email Vault recovers your Device Password. It is securely stored in Ambire's infrastructure cloud.",
```

Comment:
I think the first part should be changed to 
"The Email Vault recovery allows you to change you device password."

Reply:
I'll sync with Miro about the copy, as I get it from the design.